windows/src/engine/keyman32/appint/aiTIP.cpp

```diff
@@ -226,34 +226,39 @@ extern "C" __declspec(dllexport) BOOL WINAPI TIPProcessKey(WPARAM wParam, LPARAM
 	_td->TIPGetContext = ctfunc;
 
   AppContextWithStores *savedContext = NULL;  // I4370   // I4978
-  AppContext *savedContextUsingCore  = NULL; // used for common core
-
+  AppContext *savedContextUsingCore  = NULL;  // used for common core
+  km_kbp_option_item *SavedKBDOptions = NULL; // used for common core
   if (!Updateable) {
     if (isUsingCoreProcessor) {
       savedContextUsingCore = new AppContext();
       _td->app->CopyContext(savedContextUsingCore);
+      SavedKBDOptions = SaveKeyboardOptionsCore(_td->lpActiveKeyboard);
     } else {                                                                                   // I4370
       savedContext = new AppContextWithStores(_td->lpActiveKeyboard->Keyboard->cxStoreArray);  // I4978
       _td->app->SaveContext(savedContext);
     }
   }
+
   BOOL res = ProcessHook();
 
   if (!Updateable) {
     if (isUsingCoreProcessor) {
       if (res) {
         // Reset the context if match found
         _td->app->RestoreContextOnly(savedContextUsingCore);
+        RestoreKeyboardOptionsCore(_td->lpActiveKeyboard->lpCoreKeyboardState, SavedKBDOptions);
+        DisposeKeyboardOptionsCore(&SavedKBDOptions);
+        SavedKBDOptions = NULL;
```

Comment:
Do you want to move the second line of this into `DisposeKeyboardOptionsCore`, given we are using that pattern elsewhere?

Reply:
Yes